### PR TITLE
show update notification to promote new checkin feature.

### PR DESCRIPTION
**Pull request by simonroesch**
show notification only if enable in config service, user will see update-boarding when opening the app and time is between 8am and 8pm

### CODE REVIEW

File: app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt [hunk 1]
```diff
@@ -95,6 +98,15 @@ class ConfigWorker(context: Context, workerParams: WorkerParameters) : Coroutine
 				}
 			} else {
 				cancelNotification(context)
+
+				if (config.isCheckInUpdateNotificationEnabled && !secureStorage.checkInUpdateNotificationShown
+					&& secureStorage.onboardingCompleted && secureStorage.lastShownUpdateBoardingVersion < CHECKIN_UPDATE_BOARDIN_VERSION
+					&& !secureStorage.forceUpdateLiveData.hasObservers()
```
Comment (gallmann-ubique) [inline app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt:104]: Why is this hasObservers() check necessary?

File: app/src/main/java/ch/admin/bag/dp3t/onboarding/OnboardingSlidePageAdapter.kt [hunk 2]
```diff
@@ -20,6 +20,7 @@ class OnboardingSlidePageAdapter(fragmentActivity: FragmentActivity, onboardingT
 	companion object {
 		// Increment this number for each new Update Boarding
 		const val UPDATE_BOARDING_VERSION = 2
+		const val CHECKIN_UPDATE_BOARDIN_VERSION = 2
```
Comment (gallmann-ubique) [inline app/src/main/java/ch/admin/bag/dp3t/onboarding/OnboardingSlidePageAdapter.kt:23]: typo

File: app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt (hunk 1 above)
Comment (simonroesch) [reply app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt:104]: if you open the app (and trigger the config-request from there) we do not want to show a notification

Comment (gallmann-ubique) [reply app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt:104]: Does this work, if the onboarding activity is opened at app start? As far as I have seen, the forceUpdateLiveData is only observed by the MainActivity.

Comment (simonroesch) [reply app/src/main/java/ch/admin/bag/dp3t/networking/ConfigWorker.kt:104]: when I tested it it worked, yes. And in most cases the configrequest will happen in the background anyway and not when people open the app.

**Review by gallmann-ubique — APPROVED** [review]

**Pull request merged**